domain/schema/model/sql/0022-resource.sql

```diff
@@ -136,11 +136,10 @@ CREATE TABLE unit_resource (
 );
 
 CREATE TABLE resource_oci_image_metadata_store (
-    resource_uuid TEXT NOT NULL,
+    uuid TEXT NOT NULL,
+    storage_key TEXT NOT NULL,
```

Comment:
I've discussed this with @manadart and put the outcomes of the discussion [here](https://github.com/juju/juju/pull/18372#discussion_r1856674442). It would be good to get his opinion on precisely how I have implemented this, I'm not to sure that this is the best way myself.